x-pack/platform/plugins/shared/features/server/feature_registry.ts

```diff
@@ -41,6 +41,17 @@ export interface GetKibanaFeaturesParams {
   omitDeprecated?: boolean;
 }
 
+/**
+ * Describes parameters used to retrieve all Kibana features (for public consumers).
+ */
+export interface GetKibanaFeaturesParams {
```

Comment:
__note:__ Due to the way the start contract of the Feature plugin is defined, it was technically possible to pass any arguments to the `getKibanaFeatures` method, even though this was not reflected in the type definition:

```ts
getKibanaFeatures: this.featureRegistry.getAllKibanaFeatures.bind(this.featureRegistry),
```

This change explicitly defines the list of accepted arguments, ensuring that only `omitDeprecated` is exposed publicly.